### PR TITLE
Alert PR2: add native functionality and more stories

### PR DESCRIPTION
**Pull request by divyanshu013**
Adds `Alert` component native functionality and more stories.

## Screenshots

Web:

| Full bleed | Primary action only |
| - | - |
| image | image |

- [See more stories on Chromatic](https://61c19ee8d3d282003ac1d81c-qgaimlagwr.chromatic.com/?path=/docs/components-alert--default)

Native:

| Default | Description only | Full bleed |
| - | - | - |
| simulator_screenshot_54B36FAE-C808-4AF2-9CA0-F54C3C37A110 | simulator_screenshot_C11931B6-413F-4D31-878C-706CAC7BD665 | simulator_screenshot_35E95439-9925-4394-9C6A-18C75BCCDD57 |

## Task list

In PR1 https://github.com/razorpay/blade/pull/654:
- [x] Web base functionality

In this PR (PR2):

- [x] Native
- [x] More stories

Plus some review comments from PR1

In next PR (PR3) https://github.com/razorpay/blade/pull/670:
- [ ] Tests
- [ ] More icons
- [ ] Expose final export for `Alert`
- [ ] Changelog

### CODE REVIEW

**Comment by chaitanyadeorukhkar** [comment]
The alignment of icon & text seems a bit off when there is no title on small screens. This still looks okay on larger screens with smaller fonts. Is this intended?

Screenshot 2022-08-25 at 4 57 04 PM

File: packages/blade/src/components/Alert/Alert.stories.tsx [hunk 1]
```diff
@@ -88,11 +90,126 @@ const meta: Meta<AlertProps> = {
...
+};
+FullWidth.parameters = {
+  docs: {
+    description: {
+      story: 'A full width Alert can be used to span the entire width of its container',
+    },
+  },
+};
+
+export const FullBleed: ComponentStory<typeof AlertComponent> = ({ ...args }) => {
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Alert/Alert.stories.tsx:191]: Why are we calling this FullBleed? The story name doesn't seem descriptive enough. How about calling it "Description Only - Borderless"?

File: packages/blade/src/components/Alert/Alert.tsx [hunk 2]
```diff
@@ -185,17 +192,20 @@ const Alert = ({
...
   const closeButton = isDismissable ? (
-    <IconButton
-      accessibilityLabel="Dismiss alert"
-      onClick={onClickDismiss}
-      contrast={contrast}
-      size="large"
-      icon={CloseIcon}
-    />
+    <CloseButtonWrapper>
+      <IconButton
```
Comment (chaitanyadeorukhkar) [inline packages/blade/src/components/Alert/Alert.tsx:196]: Did we decide on a `hitSlop`? The icon button is pretty difficult to click 😅

**Review by chaitanyadeorukhkar — APPROVED** [review]
Lgtm

File: packages/blade/src/components/Alert/Alert.stories.tsx (hunk 1 above)
Comment (divyanshu013) [reply packages/blade/src/components/Alert/Alert.stories.tsx:191]: It's a term designers / printers generally use for edge to edge design. But, right not everyone might be aware, definitely can make it more descriptive.

File: packages/blade/src/components/Alert/Alert.tsx (hunk 2 above)
Comment (divyanshu013) [reply packages/blade/src/components/Alert/Alert.tsx:196]: This is pending design inputs tracking https://github.com/razorpay/blade/issues/639 /cc @saurav12 😬

File: packages/blade/src/components/Alert/Alert.stories.tsx [hunk 3]
```diff
@@ -88,11 +90,126 @@ const meta: Meta<AlertProps> = {
...
+};
+DescriptionOnly.parameters = {
+  docs: {
+    description: {
+      story: 'Sometimes a description is enough to set the required context',
+    },
+  },
+};
+
+export const PrimaryActionOnly = AlertTemplate.bind({});
```
[the remark this answers is not in the document]
Comment (divyanshu013) [reply packages/blade/src/components/Alert/Alert.stories.tsx:152]: It isn't a valid usecase but we're not doing any validation on this. I think I can manage a TS check for restricting this.

**Comment by divyanshu013** [comment]
Regarding the line height issue, seems like a line-height issue on smaller displays, [checked](https://razorpay.slack.com/archives/G01B3LQ9H0W/p1661428961958789?thread_ts=1661403647.341549&cid=G01B3LQ9H0W) with @saurav12. He mentioned we should tackle that later.

File: packages/blade/src/components/Alert/Alert.stories.tsx (hunk 1 above)
Comment (divyanshu013) [reply packages/blade/src/components/Alert/Alert.stories.tsx:191]: Updated in https://github.com/razorpay/blade/pull/670

File: packages/blade/src/components/Alert/Alert.stories.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (divyanshu013) [reply packages/blade/src/components/Alert/Alert.stories.tsx:152]: Done in https://github.com/razorpay/blade/pull/670, think that should suffice

File: packages/blade/src/components/Alert/Alert.tsx [hunk 4]
```diff
@@ -111,7 +117,7 @@ const intentIconMap = {
   notice: AlertIcon,
 };
 
-const Alert = ({
+export const Alert = ({
```
Comment (anuraghazra) [inline packages/blade/src/components/Alert/Alert.tsx:120]: Let's export at the end, easier to track what we are exporting.

Comment (anuraghazra) [reply packages/blade/src/components/Alert/Alert.tsx:120]: Export the AlertProps as well.

File: packages/blade/src/components/Alert/StyledAlert.d.ts [hunk 5]
```diff
@@ -1,13 +1,13 @@
 import type { ReactNode } from 'react';
 
-import type { ColorContrast, IntentTypes } from '~tokens/theme/theme';
+import type { ColorContrast, Feedback } from '~tokens/theme/theme';
 
 export type StyledAlertProps = {
   children: ReactNode[];
```
Comment (anuraghazra) [inline packages/blade/src/components/Alert/StyledAlert.d.ts:6]: Pretty sure you don't need to define ReactNode as array. It already has `ReactNodeArray` defined.

**Review by anuraghazra — APPROVED** [review]
Mostly LGTM

File: packages/blade/src/components/Alert/StyledAlert.d.ts (hunk 5 above)
Comment (divyanshu013) [reply packages/blade/src/components/Alert/StyledAlert.d.ts]: Done in https://github.com/razorpay/blade/pull/666/commits/f2ae55867de2cc17e9f039fa0cd99e49df089d25

File: packages/blade/src/components/Alert/Alert.tsx (hunk 4 above)
Comment (divyanshu013) [reply packages/blade/src/components/Alert/Alert.tsx]: Done in https://github.com/razorpay/blade/pull/666/commits/f2ae55867de2cc17e9f039fa0cd99e49df089d25

**Pull request merged**